src/Sylius/Bundle/ShopBundle/Grid/ProductGrid.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShopBundle\Grid;
+
+use Sylius\Bundle\GridBundle\Builder\Field\DateTimeField;
+use Sylius\Bundle\GridBundle\Builder\Field\Field;
+use Sylius\Bundle\GridBundle\Builder\Field\StringField;
+use Sylius\Bundle\GridBundle\Builder\Filter\Filter;
+use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Bundle\GridBundle\Grid\AbstractGrid;
+use Sylius\Bundle\GridBundle\Grid\ResourceAwareGridInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Webmozart\Assert\Assert;
+
+final class ProductGrid extends AbstractGrid implements ResourceAwareGridInterface
+{
+    /**
+     * @param TaxonRepositoryInterface<TaxonInterface> $taxonRepository
+     */
+    public function __construct(
+        private string $resourceClass,
+        private ChannelContextInterface $channelContext,
+        private LocaleContextInterface $localeContext,
+        private TaxonRepositoryInterface $taxonRepository,
+        private RequestStack $requestStack,
+        private bool $includeAllDescendants,
+    ) {
+    }
+
+    public static function getName(): string
+    {
+        return 'sylius_shop_product';
+    }
+
+    public function buildGrid(GridBuilderInterface $gridBuilder): void
+    {
+        $request = $this->requestStack->getMainRequest();
+        Assert::notNull($request, 'No main request available.');
+
+        $localeCode = $this->localeContext->getLocaleCode();
+
+        $slug = $request->attributes->get('slug', '');
```

Comment:
```patch
---    $slug = $request->attributes->get('slug', '');
---    if (!is_string($slug)) {
---        throw new NotFoundHttpException('The request attribute "slug" must be a string.');
---     }
+++     $slug = $request->attributes->getString('slug');
        $taxon = $this->taxonRepository->findOneBySlug($slug, $localeCode);
        if ($taxon === null) {
            throw new NotFoundHttpException('Requested page is invalid');
        }
```
`getString` will enforce type and already has `''` fallback.

![image](https://github.com/Sylius/Sylius/assets/870747/4b3901eb-d9a0-4e71-8eae-f8a536337809)

Reply:
Nice, didn't know that existed.